File: org.kevoree.modeling.microframework/src/main/java/org/kevoree/modeling/api/data/AbstractDataStore.java
package org.kevoree.modeling.api.data;

import org.kevoree.modeling.api.events.ModelElementListener;
import org.kevoree.modeling.api.events.ModelEvent;

public abstract class AbstractDataStore implements DataStore {

    private final EventDispatcher selector = new EventDispatcher();

    @Override
    public void notify(ModelEvent event) {
        selector.dispatch(event);
    }

    @Override
    public void register(ModelElementListener listener, long from, long to, String path) {
        selector.register(listener, from, to, path);
    }

    @Override
    public void unregister(ModelElementListener listener) {
        selector.unregister(listener);
    }

}

File: org.kevoree.modeling.microframework/src/main/java/org/kevoree/modeling/api/persistence/AbstractDataStore.kt
package org.kevoree.modeling.api.persistence

import org.kevoree.modeling.api.events.ModelElementListener
import org.kevoree.modeling.api.events.ModelEvent

/**
 * Created by gregory.nain on 05/09/2014.
 */


public abstract class AbstractDataStore : DataStore {

    private val selector = EventDispatcher()

    override fun register(listener: ModelElementListener, from: Long?, to: Long?, path: String) {
        selector.register(listener, from, to, path)
    }

    override fun unregister(listener: ModelElementListener) {
        selector.unregister(listener)
    }

    override fun notify(event: ModelEvent) {
        selector.dispatch(event)
    }

}



